aries-site/src/examples/templates/FilterControls/FilterContext.js

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 - Hewlett Packard Enterprise Company
+
+import { createContext, useState } from 'react';
+
+const useFilter = () => {
```

Comment:
Any reason to call is useFilter here but then export as FilterContext below? Seems like we could just call this FilterContext

Reply:
Re-worked this. Introduced a useFilters() hook and a FiltersProvider. 

From a caller point of view, they now only have to:
```
const filtersConfig = [
    { property: 'role', label: 'Role', filterType: 'CheckBoxGroup' },
    { property: 'status', label: 'Status', filterType: 'CheckBoxGroup' },
    {
      property: 'location',
      label: 'Location',
      filterType: 'CheckBoxGroup',
    },
    {
      property: 'hoursAvailable',
      label: 'Remaining Hours Available',
      filterType: 'RangeSelector',
      inputProps: {
        min: 0,
        max: 40,
        valueRange: '0 - 40 hours',
      },
    },
    { property: 'name', label: 'Name', filterType: 'CheckBoxGroup' },
  ];

  return (
    <>
      <Heading level={2} margin={{ bottom: 'small', top: 'none' }}>
        Users
      </Heading>
      <FiltersProvider>
        <FilterControls data={users} filters={filtersConfig} />
        <Users />
      </FiltersProvider>
    </>
  );
```

The rest of the work is handled by the FilterControls components.